### ISSUE TITLE
Add option for queued download

### ISSUE THREAD

**Issue by jens-maus** [jens-maus/yam#230]
**Originally by _mario@cattaneo.de_ on 2010-11-07 14:00:39 +0100**

___
YAM 2.7-dev [OS4/PPC] (06.11.2010) Copyright (C) 2000-2010 YAM Open Source Team

I know "hate" is a hard word, but as much as I like that YAM starts tomultitake, I hat it that the the multi-threaded-pop does mix up the mails in my inbox. I liked it to have the way, that they are listed, one account after the other (I don't want to add filtering in different folder, especially as I don't want different folders. ;) It was just convinient, when checking "what's new" to process them "account by account". An option for this would be nice. This would not mean, that pop-has to block YAM again, just have the POPs be patiened and queued. :)

**Comment by tboeckel** [comment]
**Originally on 2010-11-08 08:19:41 +0100**

___
The way how YAM sorts the mails in the inbox has changed in no way. The only difference is that YAM now downloads the mails in parallel instead of one account after another. But this should not influence the sorting.

**Comment by jens-maus** [comment]
**Originally by _mario@cattaneo.de_ on 2010-11-09 16:47:02 +0100**

___
Well, my sorting order is "date received" and that of course differs for mixed or "queued" download. :)

**Comment by tboeckel** [comment]
**Originally on 2010-11-09 16:54:03 +0100**

___
Although YAM offers the possibility to let the mails be sorted by receive date I really suggest to switch that to compose date. The receive date is something very specific to your system (as you now noticed), while the compose date is more or less absolute. Furthmore this will give you the true order in which the mails were originally created, no matter in which order or when you received them.

**Comment by jens-maus** [comment]
**Originally by _mario@cattaneo.de_ on 2010-11-09 17:08:59 +0100**

___
You don't seem to understand, what I mean.
I liked it the way it was (for years) it gave me a nice "account by account" order. When processing the mails, I always knew "which account I was in" and how to intuitivly handle each mail.
This is my personal taste.
If a queued download can/will not be offered, just close here and set to "won't fix". It's better than trying to tell me, that I use YAM the wrong way... - there is no right or wrong way, it's all about taste and preference.

**Comment by tboeckel** [comment]
**Originally on 2010-11-10 12:33:12 +0100**

___
I cannot really tell if queued download can or cannot be made possibe. Currently it is not possible due to how the POP3 access is implemented.

Another approach would be to add an additional level in the folder tree. This would give you separate set of standard folders (in, out, sent, trash, spam) for each account. With this approach a queued download as you would like it is no longer necessary as the mails will be downloaded to separate folders and hence you get the same behaviour as you like it. Furthermore, from my point of view, this approach is absolutely necessary, because otherwise the support for multiple SMTP servers is impossible (how should YAM know which mail in the outgoing folder is to be sent via which SMTP server?).

To make it short, I won't close this ticket as "won't fix", but delay it until YAM 2.8 as I think the changes required are by far too many to finish them within a sensible time frame for YAM 2.7.

**Comment by jens-maus** [comment]
**Originally on 2012-06-02 02:18:27 +0200**

___
Have to move that item to 2.9 as this ticket would require an .index change which is not scheduled for YAM 2.8. In addition Thore and myself have consens in that we are going to add a new column "Received Mail Server" in the main mail list which then would allow to sort by received mail server and the first sort option and then by date received. This would then hopefully close that ticket.

**Comment by jens-maus** [comment]
**Originally on 2013-11-04 20:54:44 +0100**

___
In (57e97e0):
 * MainMailList.c, YAM_MAf.c, tcp/pop3.c, misc: implemented a new
  "Mail Account" column in the main mail list which will show the mail
  account as "username@hostname" over which the particular mail had
  been transferred. This new column should allow to sort emails based
  on mail account/hostname a mail was received so that users can better
  order their mails due to the fact that since YAM 2.8 mails are now
  transferred asynchronously from multiple servers and thus mails from
  different mail accounts end up mixed in the incoming folder. 
  This closes #230.

**Comment by jens-maus** [comment]
**Originally on 2013-12-22 19:13:32 +0100**

___
**Release Note:**
implemented a new "Mail Account" column in the main mail list to be able to sort emails based on mail account/hostname a mail was received so that users can better order their mails due to the fact that since YAM 2.8 mails are now transferred asynchronously from multiple servers.
